### ISSUE TITLE
[cli] Wait / wait-state flags do not work as expected

### ISSUE THREAD

**Issue by kelchm** [rancher/rancher#8492]
**Rancher Versions:**
cli: v0.4.1

**Steps to Reproduce:**
Execute a rancher up command similar to this one:
`rancher --wait --wait-state "healthy" up -s "my-stack-123" --pull -d`

**Results:** 
The rancher cli returns with a success code while the `healthState` of the stack is still `initializing`.

**Comment by bastiandg** [comment]
Hey.
We stumbled upon the same issue. After a bit of investigation we found the core of the Issue:
The function [NewWaiter](https://github.com/rancher/cli/blob/master/cmd/wait.go#L52) which implements the `--wait*` parameters isn't used in the [up](https://github.com/rancher/cli/blob/master/cmd/up.go) command.

So instead of using `rancher --wait --wait-state "healthy" up` you have split the command up into two:

`rancher up -s "my-stack-123" --pull -d`
`rancher --wait-state "healthy" wait my-stack-123`

Nonetheless this is an issue. `--wait` is advertised a [global option](https://docs.rancher.com/rancher/v1.6/en/cli/commands/#rancher-cli-global-options) - which it isn't.

**Comment by deniseschannon** [comment]
The issue with @StrongMonkey's approach is that it works for the first `rancher up` but not any subsequent call.

**Comment by cjellick** [comment]
@amy please update this issue with all the cli scenarios you tested and how

**Comment by amy** [comment]
Please use the following scenarios for manual testing:

Failure Stack:
```
// docker-compose.yml
version: '2'
services:
  amy-test:
    image: nginx
    scale: 1
    health_check:
      port: 80
      request_line: GET /fail HTTP/1.0
      interval: 2000
      initializing_timeout: 60000
      reinitializing_timeout: 60000
      unhealthy_threshold: 5
      strategy: none
      healthy_threshold: 2
      response_timeout: 2000
```
Success Stack:
```
// docker-compose.yml
version: '2'
services:
  amy-test:
    image: nginx
    restart: always
    scale: 1
```
Combined Stack:
```
// docker-compose.yml
version: '2'
services:
  amy-failure:
    image: nginx
    scale: 1
    health_check:
      port: 80
      request_line: GET /fail HTTP/1.0
      interval: 2000
      initializing_timeout: 60000
      reinitializing_timeout: 60000
      unhealthy_threshold: 5
      strategy: none
      healthy_threshold: 2
      response_timeout: 2000
  amy-success:
    image: nginx
    scale: 1
    health_check:
      port: 80
      request_line: GET /fail HTTP/1.0
      interval: 2000
      initializing_timeout: 60000
      reinitializing_timeout: 60000
      unhealthy_threshold: 5
      strategy: none
      healthy_threshold: 2
      response_timeout: 2000
```

In testing, the above docker-compose files correlate with the stack name in the command, so if 
the stack name is `failure-stack` in the command, please use the failure stack docker-compose file.
NOTE: Make sure to delete stacks in between scenarios unless otherwise indicated

Scenario 0: Failure
`rancher --wait --wait-state "initializing" up -s "failure-stack" --pull -d`
    Behavior: Cli should return and the state of the stack should be initializing.

Scenario 1: Failure
`rancher --wait --wait-state "healthy" up -s "failure-stack" --pull -d`
    Behavior: Cli should not return.

Scenario 2: Failure Timeout
`rancher --wait --wait-state "healthy" --wait-timeout 10 -s "failure-stack" --pull -d`
    Behavior: Cli should return after timeout period.

Scenario 3: Failure wait
`rancher --wait "failure-stack" --pull -d`
    Behavior: Cli should return.

Scenario 4: Success
rancher --wait --wait-state "healthy" up -s "success-stack" --pull -d
    Behavior: Stack should return. And state of stack in the UI should be healthy.

Scenario 5: Success
`rancher --wait up -s "success-stack" --pull -d`
    Behavior: Stack should return. And state of stack in the UI should be healthy.

Scenario 6: Failure & Success
`rancher up -s "failure-stack" --pull -d`
`rancher --wait --wait-state "healthy" up -s "success-stack" --pull -d`
    Behavior: Stack should return. And state of stack in the UI should be healthy.

Scenario 7: Success & Failure
`rancher up -s "success-stack" --pull -d`
`rancher --wait --wait-state "healthy" up -s "failure-stack" --pull -d`
    Behavior: Stack should not return.

Scenario 8: Combined
`rancher --wait --wait-state "healthy" up -s "combined-stack" --pull -d`
    Behavior: Stack should not return.

Scenario 9: Combined
`rancher --wait --wait-state "healthy" up --wait-timeout 10 -s "combined-stack" --pull -d`
    Behavior: Stack should return after timeout.

Scenario 10: Combined
`rancher --wait "combined-stack" --pull -d`
    Behavior: Stack should return.

**Comment by amy** [comment]
Available in cli version `v0.6.4-rc1`

**Comment by soumyalj** [comment]
Tested on rancher v1.6.8-rc1 with CLI version v0.6.4-rc1. wait/wait-state flags are functional. All the examples listed above succeeded.
